src/pykeen/datasets/analysis.py

```diff
@@ -191,13 +191,13 @@ def entity_relation_co_occurrence_dataframe(dataset: Dataset) -> pd.DataFrame:
 
 
 def _get_skyline(xs: Collection[Tuple[int, float]]) -> Collection[Tuple[int, float]]:
-    # TODO: naive implementation, O(n2)
+    xs = sorted(xs)
     return {
         (s, c)
-        for s, c in xs
+        for i, (s, c) in enumerate(xs)
         if not any(
-            s2 > s and c2 > c
-            for s2, c2 in xs
+            s2 > s and c2 >= c
+            for s2, c2 in xs[i:]
```

Comment:
Still `O(n^2)` isn't it? :smile: But besides, might be an improvement in runtime.

Reply:
yes, true. but this will half the runtime ;)

I have been staring at this for like 80 minutes trying to come up with something better. I know there are algorithms for this. They're totally non-intuitive but super elegant. It's a reminder I can't come up with clever algorithms for 💩  these days

Reply:
https://stackoverflow.com/questions/19059878/dominant-set-of-points-in-on

Reply:
I think this one is applicable, cf. db2c4fd